phase1-coordinator/src/coordinator_state.rs

```diff
@@ -2323,30 +2417,86 @@ impl CoordinatorState {
     ///
     /// On success, returns a list of justifications for the coordinator to take actions on.
     ///
-    #[inline]
-    pub(super) fn update_dropped_participants(&mut self) -> Result<Vec<Justification>, CoordinatorError> {
+    pub(super) fn update_dropped_participants(
+        &mut self,
+        time: &dyn TimeSource,
+    ) -> Result<Vec<Justification>, CoordinatorError> {
+        Ok(self
+            .update_contributor_seen_drops(time)?
+            .into_iter()
+            .chain(self.update_participant_lock_drops(time)?.into_iter())
+            .collect())
+    }
+
+    /// This will drop a participant (verifier or contributor) if it
+    /// has been holding a lock for longer than
+    /// [crate::environment::Environment]'s
+    /// `participant_lock_timeout`.
+    fn update_participant_lock_drops(&mut self, time: &dyn TimeSource) -> Result<Vec<Justification>, CoordinatorError> {
         // Fetch the timeout threshold for contributors.
-        let contributor_timeout = self.environment.contributor_timeout_in_minutes() as i64;
+        let participant_lock_timeout = self.environment.participant_lock_timeout();
 
         // Fetch the current time.
-        let now = Utc::now();
+        let now = time.utc_now();
 
-        // Initialize the list of justifications.
-        let mut justifications = vec![];
+        self.current_contributors
+            .clone()
+            .iter()
+            .chain(self.current_verifiers.clone().iter())
+            .filter_map(|(participant, participant_info)| {
+                dbg!(participant);
+                dbg!(&participant_info.locked_chunks);
```

Comment:
probably going to be removed before merging?

Reply:
good spot, thanks I'll remove!